consul/snapshot_endpoint.go

```diff
@@ -101,10 +101,34 @@ func (s *Server) dispatchSnapshotRequest(args *structs.SnapshotRequest, in io.Re
 			return nil, err
 		}
 
+		// This'll be used for feedback from the leader loop.
+		errCh := make(chan error, 1)
+		timeoutCh := time.After(time.Minute)
+
 		select {
 		// Tell the leader loop to reassert leader actions since we just
 		// replaced the state store contents.
-		case s.reassertLeaderCh <- struct{}{}:
+		case s.reassertLeaderCh <- errCh:
+
+		// We might have lost leadership while waiting to kick the loop.
+		case <-timeoutCh:
```

Comment:
Was thinking an overall minute for queue and run vs. restarting another minute here. It's pretty arbitrary though.